controllers/release/adapter.go

```diff
@@ -69,6 +69,39 @@ func newAdapter(ctx context.Context, client client.Client, release *v1alpha1.Rel
 	}
 }
 
+// EnsureAuthorIsValid is an operation that will ensure that Release is attributed to a specific,
+// valid user.
+func (a *adapter) EnsureAuthorIsValid() (controller.OperationResult, error) {
+	if !a.release.IsReleasing() {
+		return controller.ContinueProcessing()
+	}
+
+	if a.release.Labels[metadata.AutomatedLabel] == "true" && !a.release.IsAutomated() {
+		return controller.RequeueWithError(fmt.Errorf("automated not set in status for automated release"))
+	}
+
+	patch := client.MergeFrom(a.release.DeepCopy())
+	releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
This error handling could be merged with the one in L96 so we don't duplicate that much code. This whole validation thing is so annoying... I'm really considering to follow the approach of the validate functions with the same signature so we can loop over them. We can pass the resources to all of them and return an error. At the point of validate stuff, the processing resources and the release is everything a validation function could need.

Reply:
how? we don't want to call line 95 unless the error is not nil from line 84

Reply:
Changing it completely ^^